ext/posix/sys/socket.c

```diff
@@ -386,7 +395,7 @@ Pconnect(lua_State *L)
 	int fd = checkint(L, 1);
 	checknargs (L, 2);
 	if (sockaddr_from_lua(L, 2, &sa, &salen) != 0)
-		return pusherror(L, "not a valid IPv4 dotted-decimal or IPv6 address string");
```

Comment:
I'm curious why you don't like the existing error message?

Reply:
Because error may be raised not only by IP syntax, but also due to wrong interface name, or other fields in the table.